### ISSUE TITLE
[Efficiency] User encounters unnecessary repetition or redundancy.

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#75979]
_This ticket was generated during the pilot process for the new VA.gov experience standards. Please tag `@platform-governance-team-members` on Slack if you need assistance interpreting this ticket._

## Product Information
**Team:** govcio-vfep
**Product:** GI Bill Comparison Tool
**Feature:** Specialized Mission Filter

## Findings details
**VA.gov Experience Standard - issue:** User encounters unnecessary repetition or redundancy.
**VA.gov Experience Standard - category:** Efficiency
**Launch-blocking:** No
**Collab Cycle Reviewer:** @briandeconinck (Accessibility)

### Description

In each of the fieldsets of checkboxes, the heading for that fieldset is announced by screen readers before every checkbox. This repetition is not needed.

When a screen reader user tabs into a group of checkboxes or radio buttons, the expected behavior typically goes like this:
- When the first checkbox to receives focus, the legend text is announced, then the label for the first checkbox.
- When each subsequent checkbox receives focus, just the label of that checkbox is announced.

This pattern provides appropriate context (these checkboxes are grouped together under this legend) without being too repetitive. Unnecessary repetition of text can be a major annoyance for screen reader users.

<details><summary>Link, screenshot or steps to recreate</summary>

Steps to reproduce:

1. Navigate to the [GI Bill Comparison Tool on staging](https://staging.va.gov/education/gi-bill-comparison-tool/).
2. Activate a screen reader (recommend VoiceOver on Mac or NVDA on Windows for testing).
3. Press [tab] until the checkbox labeled "Public" receives focus. Note that the screen reader announces the heading as part of its message, eg. on NVDA: `School types, Public, checkbox unchecked`. This matches expected behavior.
4. Press [tab] one time to move to the next checkbox. Note that the screen reader continues to announce the heading, eg. `School types, For profit, checkbox unchecked`. This (and every subsequent time it announces "School types") does not match expected behavior.

</details>

## Recommended action
This is happening because each checkbox references the element that contains the heading and the link via `aria-labelledby`, eg:
```
   <span id="checkbox-group-85-legend" class="gibct-legend">
      <h3 class="school-types-label" aria-level="2">School types</h3>
   </span>
```
and
```
   <div class="vads-l-col--1p5 about-school-checkbox">
      <input id="checkbox-group-93-0" name="PUBLIC" data-testid="school-type-Public" type="checkbox" aria-labelledby="checkbox-group-93-legend public-0-label">
      <label class="gi-checkbox-label vads-u-margin-right--3" id="public-0-label" name="PUBLIC-label" for="checkbox-group-93-0">Public</label>
   </div>
```

I think this could largely be solved via a more typical `fieldset`/`legend` pattern rather than using `aria-labelledby`, along the lines of this:
```
<fieldset>
   <legend><h3>School types</h3></legend>

   <input id="checkbox-group-93-0" name="PUBLIC" data-testid="school-type-Public" type="checkbox" />
   <label for="checkbox-group-93-0">Public</label>

   <input id="checkbox-group-97-1" name="FOR PROFIT" data-testid="school-type-For profit" type="checkbox" />
   <label for="checkbox-group-97-1">For profit</label>
    
   ...
</fieldset>  
```

This would result in the expected screen reader behavior: When tabbing into the fieldset the first time, the legend is announced followed by the label of the checkbox receiving focus. When tabbing to subsequent checkboxes in the fieldset, only the label of the checkbox would be announced. Screen reader users would understand from context that these checkboxes are grouped together.

If there are technical limitations that prevent you from using the va-checkbox-group component, please use that component as a guide for how to structure your HTML.

## References

- **WCAG Success Criteria:** [WCAG 2.5.3 Label in Name - Level A](https://www.w3.org/TR/WCAG21/#label-in-name)
- **Modality:** N/A
- **Design System Component:** N/A
- **Design System Pattern:** N/A
- **Content Style Guide:** N/A
- **Context:** N/A
- **Accessibility Defect Severity:** [3: Medium. Should be fixed in 1-3 sprints post-launch.](https://depo-platform-documentation.scrollhelp.site/developer-docs/accessibility-defect-severity-rubric#Accessibilitydefectseverityrubric-a11y-defect-3)

---

### Next Steps for the VFS Team

- [ ] **Questions?** For the most timely response, comment on Slack in your team channel tagging `@platform-governance-team-members` with any questions or to get help validating the issue.
- [ ] **Ticket ownership:** Platform OCTO leadership asks that we document all issues we identify. If you believe that this issue is out of scope or not your team's responsibility, it is still your responsibility to resolve the issue. Please comment on Slack in your team channel tagging `@platform-governance-team-members`, providing an explanation and who you believe is responsible. Governance team will research and follow up.
- [ ] **Close this ticket** when the issue has been resolved or validated by your Product Owner.
